### PR TITLE
Push/remove command line args for mobile

### PR DESCRIPTION
**Pull request by improbable-valentyn**
#### Description
Merge remaining functionality related to pushing and removing command line from mobile devices.
Android command line file will be automatically removed after a launch on device session to prevent it from overriding command line that's built into the APK.

Original PRs: https://github.com/spatialos/UnrealGDK/pull/2016 https://github.com/spatialos/UnrealGDK/pull/2132
Original authors: @jayprobable @yujunting 

#### Primary reviewers
@jessicafalk @MatthewSandfordImprobable

### CODE REVIEW

File: SpatialGDK/Source/SpatialGDKEditor/Public/SpatialGDKEditorCommandLineArgsManager.h [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+#ifdef ENABLE_LAUNCHER_DELEGATE
+	void OnCreateLauncher(ILauncherRef LauncherRef);
+	void OnLaunch(ILauncherWorkerPtr LauncherWorkerPtr, ILauncherProfileRef LauncherProfileRef);
+	void OnLauncherCanceled(double ExecutionTime);
+	void OnLauncherFinished(bool Outcome, double ExecutionTime, int32 ReturnCode);
+
+	void RemoveCommandLineFromDevice();
+#endif // ENABLE_LAUNCHER_DELEGATE
+
+	static bool TryConstructMobileCommandLineArgumentsFile(FString& CommandLineArgsFile);
```
Comment (MatthewSandfordImprobable) [inline SpatialGDK/Source/SpatialGDKEditor/Public/SpatialGDKEditorCommandLineArgsManager.h:41]: Could prefixed  CommandLineArgsFile prefixed with "Out"

File: SpatialGDK/Source/SpatialGDKEditor/Public/SpatialGDKEditorCommandLineArgsManager.h [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+	static FReply PushCommandLineToIOSDevice();
+	static FReply PushCommandLineToAndroidDevice();
+	static FReply RemoveCommandLineFromAndroidDevice();
+
+private:
+#ifdef ENABLE_LAUNCHER_DELEGATE
+	void OnCreateLauncher(ILauncherRef LauncherRef);
+	void OnLaunch(ILauncherWorkerPtr LauncherWorkerPtr, ILauncherProfileRef LauncherProfileRef);
+	void OnLauncherCanceled(double ExecutionTime);
+	void OnLauncherFinished(bool Outcome, double ExecutionTime, int32 ReturnCode);
```
Comment (MatthewSandfordImprobable) [inline SpatialGDK/Source/SpatialGDKEditor/Public/SpatialGDKEditorCommandLineArgsManager.h]: bOutcome

File: SpatialGDK/Source/SpatialGDKEditor/Private/SpatialGDKEditorCommandLineArgsManager.cpp [hunk 3]
```diff
@@ -0,0 +1,280 @@
...
+{
+	if (bAndroidDevice)
+	{
+		RemoveCommandLineFromAndroidDevice();
+	}
+}
+
+void FSpatialGDKEditorCommandLineArgsManager::OnLaunch(ILauncherWorkerPtr LauncherWorkerPtr, ILauncherProfileRef LauncherProfileRef)
+{
+	LauncherWorkerPtr->OnCanceled().AddRaw(this, &FSpatialGDKEditorCommandLineArgsManager::OnLauncherCanceled);
```
Comment (jessicafalk) [inline SpatialGDK/Source/SpatialGDKEditor/Private/SpatialGDKEditorCommandLineArgsManager.cpp:60]: I remember commenting on this in the original PR, but don't remember the response. we should unbind this when needed or only add it once. Right now we will add it with each launch, right?

Comment (improbable-valentyn) [reply SpatialGDK/Source/SpatialGDKEditor/Private/SpatialGDKEditorCommandLineArgsManager.cpp:60]: Each launch creates a new instance of `FLauncherWorker` so we don't need to unbind the callback.

File: SpatialGDK/Source/SpatialGDKEditor/Private/SpatialGDKEditorCommandLineArgsManager.cpp [hunk 4]
```diff
@@ -0,0 +1,280 @@
...
+	LauncherWorkerPtr->OnCompleted().AddRaw(this, &FSpatialGDKEditorCommandLineArgsManager::OnLauncherFinished);
+
+	bAndroidDevice = false;
+	TArray<ILauncherTaskPtr> TaskList;
+	LauncherWorkerPtr->GetTasks(TaskList);
+	for (const ILauncherTaskPtr& Task : TaskList)
+	{
+		if (Task->GetDesc().Contains(TEXT("android")))
+		{
+			bAndroidDevice = true;
```
Comment (MatthewSandfordImprobable) [inline SpatialGDK/Source/SpatialGDKEditor/Private/SpatialGDKEditorCommandLineArgsManager.cpp:70]: Can we break after setting bAndroidDevice

Comment (MatthewSandfordImprobable) [reply SpatialGDK/Source/SpatialGDKEditor/Private/SpatialGDKEditorCommandLineArgsManager.cpp:70]: If so maybe we move the log line "Android device launched" here as well?

File: SpatialGDK/Source/SpatialGDKEditor/Private/SpatialGDKEditorCommandLineArgsManager.cpp [hunk 5]
```diff
@@ -0,0 +1,280 @@
...
+FReply FSpatialGDKEditorCommandLineArgsManager::GenerateDevAuthToken()
+{
+	FString DevAuthToken;
+	FString ErrorMessage;
+	if (!SpatialCommandUtils::GenerateDevAuthToken(GetMutableDefault<USpatialGDKSettings>()->IsRunningInChina(), DevAuthToken, ErrorMessage))
+	{
+		FMessageDialog::Open(EAppMsgType::Ok, FText::FromString(ErrorMessage));
+		return FReply::Unhandled();
+	}
+	if (USpatialGDKEditorSettings* SpatialGDKEditorSettings = GetMutableDefault<USpatialGDKEditorSettings>())
```
Comment (MatthewSandfordImprobable) [inline SpatialGDK/Source/SpatialGDKEditor/Private/SpatialGDKEditorCommandLineArgsManager.cpp:247]: Nit: We check SpatialGDKEditorSettings  is valid but don't check SpatialGDKSettings. Just interesting we handle them differently.

File: SpatialGDK/Source/SpatialGDKEditor/Private/SpatialGDKEditorCommandLineArgsManager.cpp [hunk 6]
```diff
@@ -0,0 +1,277 @@
...
+
+	bAndroidDevice = false;
+	TArray<ILauncherTaskPtr> TaskList;
+	LauncherWorkerPtr->GetTasks(TaskList);
+	for (const ILauncherTaskPtr& Task : TaskList)
+	{
+		if (Task->GetDesc().Contains(TEXT("android")))
+		{
+			bAndroidDevice = true;
+			UE_LOG(LogSpatialGDKEditorCommandLineArgsManager, Log, TEXT("Android device launched"));
```
Comment (jessicafalk) [inline SpatialGDK/Source/SpatialGDKEditor/Private/SpatialGDKEditorCommandLineArgsManager.cpp:71]:
```suggestion
			UE_LOG(LogSpatialGDKEditorCommandLineArgsManager, Log, TEXT("Launched game on Android device"));
```
?

**Pull request merged**